src/fake_rtc.c

```diff
@@ -56,27 +65,49 @@ void FakeRtc_AdvanceTimeBy(u32 hours, u32 minutes, u32 seconds)
 
     while(hours >= HOURS_PER_DAY)
     {
-        time->days++;
+        days++;
         hours -= HOURS_PER_DAY;
     }
 
-    time->seconds = seconds;
-    time->minutes = minutes;
-    time->hours = hours;
+    rtc->second = seconds;
+    rtc->minute = minutes;
+    rtc->hour = hours;
+
+    while (days > 0)
+    {
+        u32 remainingDaysInMonth = (sNumDaysInMonths[rtc->month - 1] + (rtc->month == MONTH_FEB && IsLeapYear(rtc->year)) - rtc->day);
+
+        if (days > remainingDaysInMonth)
+        {
+            rtc->day = 1;
+            rtc->month++;
+            if (rtc->month > MONTH_DEC)
+            {
+                rtc->month = MONTH_JAN;
+                rtc->year++;
+            }
+            days -= (remainingDaysInMonth + 1);
+            rtc->dayOfWeek = (rtc->dayOfWeek + remainingDaysInMonth + 1) % WEEKDAY_COUNT;
+        }
+        else
+        {
+            rtc->day += days;
+            rtc->dayOfWeek = (rtc->dayOfWeek + days) % WEEKDAY_COUNT;
+            days = 0;
+        }
+    }
 }
```

Comment:
Github is being stupid but I'm trying to leave a suggestion to replace this function with

```c
void FakeRtc_AdvanceTimeBy(u32 days, u32 hours, u32 minutes, u32 seconds)
{
    struct SiiRtcInfo *rtc = FakeRtc_GetCurrentTime();
    FakeRtc_AdvanceSeconds(rtc, &days, &hours, &minutes, &seconds);

    while (days > 0)
        FakeRtc_AdvanceDays(rtc, &days);
}
```